org-ai-openai-image-query.el

```diff
@@ -0,0 +1,124 @@
+;;; org-ai-openai-image-query.el --- Send base64 or URL image and question to OpenAI -*- lexical-binding: t; -*-
+
+;;; Code:
+
+;; TODO handle multiple images https://platform.openai.com/docs/guides/vision/multiple-image-inputs
+;; TODO compress images if larger https://platform.openai.com/docs/guides/vision/managing-images, https://platform.openai.com/docs/guides/vision/is-there-a-limit-to-the-size-of-the-image-i-can-upload
+;; TODO ensure file is png, jpeg, webp, or non-animated gif https://platform.openai.com/docs/guides/vision/what-type-of-files-can-i-upload
+(require 'org-ai-openai)
+(require 'org-ai-useful)
+
+(defcustom org-ai-image-query-model "gpt-4o-mini"
+  "The model to use for image queries."
+  :group 'org-ai
+  :type '(choice (const :tag "gpt-4o-mini" "gpt-4o-mini")
+          (const :tag "gpt-4o" "gpt-4o")
+          (const :tag "gpt-4-turbo" "gpt-4-turbo")))
+
+(defcustom org-ai-query-image-file nil
```

Comment:
@rksm you probably recognize this from org-ai-on-region-file, just calling attention in case you want me to do this in a different way